DependencyInjection/Configuration.php

```diff
@@ -377,25 +377,53 @@ private function addSharedPluginNodes(ArrayNodeDefinition $pluginNode, $disableA
                         ->thenInvalid('You can\'t provide config option "respect_cache_headers" and "respect_response_cache_directives" simultaniously. Use "respect_response_cache_directives" instead.')
                         ->end()
                         ->children()
-                            ->scalarNode('default_ttl')->defaultValue(0)->end()
+                            ->scalarNode('cache_lifetime')
+                                ->info('The minimum time we should store a cache item')
+                            ->end()
+                            ->enumNode('cache_key_generator')
+                                ->info('A class implementing Http\Client\Common\Plugin\Cache\Generator\CacheKeyGenerator to generate a PSR-6 cache key')
+                                ->values([null, 'Http\Client\Common\Plugin\Cache\Generator\CacheKeyGenerator'])
```

Comment:
please use CacheKeyGenerator::class instead (we are php 5.5 or later so its supported)